Flow Trigger Precondition caused Kestra to shut down if ID missing

### Describe the issue

Flows used:
```yaml
id: bronze_layer
namespace: company.team

tasks:
  - id: hello
    type: io.kestra.plugin.core.log.Log
    message: Hello World! 🚀
```

```yaml
id: silver_layer
namespace: company.team

tasks:
  - id: transform_data
    type: io.kestra.plugin.core.log.Log
    message: deduplication, cleaning, and minor aggregations

triggers:
  - id: flow_trigger
    type: io.kestra.plugin.core.trigger.Flow
    preconditions:
      timeWindow:
        type: DAILY_TIME_DEADLINE
        deadline: "15:00:00"
      flows:
        - namespace: company.team
          flowId: bronze_layer
          states: [SUCCESS]
```


When executing `bronze_layer` and then navigating to `silver_layer`, Kestra shut down and this error occured (the SLA condition needs to be met for the error to happen):
```
kestra-1    | 2024-12-02 14:51:10,612 INFO  jdbc-queue-Execution_0 flow.bronze_layer [tenant: null] [namespace: company.team] [flow: bronze_layer] [execution: KxMpiT06Rxb2RH9CR3aIe] Flow started
kestra-1    | 2024-12-02 14:51:11,497 INFO  worker_11    flow.bronze_layer.hello [tenant: null] [namespace: company.team] [flow: bronze_layer] [task: hello] [execution: KxMpiT06Rxb2RH9CR3aIe] [taskrun: 2bt5OJpDOCbUnec5XJRMv2] Type Log started
kestra-1    | 2024-12-02 14:51:11,505 INFO  worker_11    f.b.K.2bt5OJpDOCbUnec5XJRMv2 Hello World! 🚀
kestra-1    | 2024-12-02 14:51:11,507 INFO  worker_11    flow.bronze_layer.hello [tenant: null] [namespace: company.team] [flow: bronze_layer] [task: hello] [execution: KxMpiT06Rxb2RH9CR3aIe] [taskrun: 2bt5OJpDOCbUnec5XJRMv2] Type Log with state RUNNING completed in 00:00:00.886
kestra-1    | 2024-12-02 14:51:11,556 INFO  jdbc-queue-Execution_0 flow.bronze_layer [tenant: null] [namespace: company.team] [flow: bronze_layer] [execution: KxMpiT06Rxb2RH9CR3aIe] Flow completed with state SUCCESS in 00:00:01.884
postgres-1  | 2024-12-02 14:51:11.566 UTC [1038] ERROR:  null value in column "condition_id" of relation "multipleconditions" violates not-null constraint
postgres-1  | 2024-12-02 14:51:11.566 UTC [1038] DETAIL:  Failing row contains (company.team_silver_layer, {"end": "2024-12-02T15:00:00.000Z", "start": "2024-12-02T00:00:0..., company.team, silver_layer, null, 2024-12-02 00:00:00+00, 2024-12-02 15:00:00+00, null).
postgres-1  | 2024-12-02 14:51:11.566 UTC [1038] STATEMENT:  insert into multipleconditions ("key", "value") values ($1, cast($2 as jsonb)) on conflict ("key") do update set "value" = cast($3 as jsonb)
kestra-1    | 2024-12-02 14:51:11,579 ERROR jdbc-queue-Execution_0 .c.u.ThreadUncaughtExceptionHandlers Caught an exception in Thread[#59,jdbc-queue-Execution_0,5,main]. Shutting down.
kestra-1    | org.jooq.exception.IntegrityConstraintViolationException: SQL [insert into multipleconditions ("key", "value") values (?, cast(? as jsonb)) on conflict ("key") do update set "value" = cast(? as jsonb)]; ERROR: null value in column "condition_id" of relation "multipleconditions" violates not-null constraint
kestra-1    |   Detail: Failing row contains (company.team_silver_layer, {"end": "2024-12-02T15:00:00.000Z", "start": "2024-12-02T00:00:0..., company.team, silver_layer, null, 2024-12-02 00:00:00+00, 2024-12-02 15:00:00+00, null).
kestra-1    |   at org.jooq_3.19.14.POSTGRES_15.debug(Unknown Source)
kestra-1    |   at org.jooq.impl.Tools.translate(Tools.java:3638)
kestra-1    |   at org.jooq.impl.Tools.translate(Tools.java:3628)
kestra-1    |   at org.jooq.impl.DefaultExecuteContext.sqlException(DefaultExecuteContext.java:827)
kestra-1    |   at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:362)
kestra-1    |   at org.jooq.impl.AbstractDelegatingQuery.execute(AbstractDelegatingQuery.java:115)
kestra-1    |   at io.kestra.repository.postgres.PostgresRepository.persist(PostgresRepository.java:76)
kestra-1    |   at io.kestra.jdbc.runner.AbstractJdbcMultipleConditionStorage.lambda$save$2(AbstractJdbcMultipleConditionStorage.java:75)
kestra-1    |   at java.base/java.util.ArrayList.forEach(Unknown Source)
kestra-1    |   at io.kestra.jdbc.runner.AbstractJdbcMultipleConditionStorage.lambda$save$3(AbstractJdbcMultipleConditionStorage.java:73)
kestra-1    |   at org.jooq.impl.DefaultDSLContext.lambda$transaction$5(DefaultDSLContext.java:592)
kestra-1    |   at org.jooq.impl.DefaultDSLContext.lambda$transactionResult0$3(DefaultDSLContext.java:530)
kestra-1    |   at org.jooq.impl.Tools$3$1.block(Tools.java:6369)
kestra-1    |   at java.base/java.util.concurrent.ForkJoinPool.unmanagedBlock(Unknown Source)
kestra-1    |   at java.base/java.util.concurrent.ForkJoinPool.managedBlock(Unknown Source)
kestra-1    |   at org.jooq.impl.Tools$3.get(Tools.java:6366)
kestra-1    |   at org.jooq.impl.DefaultDSLContext.transactionResult0(DefaultDSLContext.java:578)
kestra-1    |   at org.jooq.impl.DefaultDSLContext.transactionResult(DefaultDSLContext.java:502)
kestra-1    |   at org.jooq.impl.DefaultDSLContext.transaction(DefaultDSLContext.java:591)
kestra-1    |   at io.kestra.jdbc.JooqDSLContextWrapper.lambda$transaction$1(JooqDSLContextWrapper.java:58)
kestra-1    |   at dev.failsafe.Functions.lambda$toCtxSupplier$11(Functions.java:243)
kestra-1    |   at dev.failsafe.Functions.lambda$get$0(Functions.java:46)
kestra-1    |   at dev.failsafe.internal.RetryPolicyExecutor.lambda$apply$0(RetryPolicyExecutor.java:74)
kestra-1    |   at dev.failsafe.internal.FallbackExecutor.lambda$apply$0(FallbackExecutor.java:51)
kestra-1    |   at dev.failsafe.SyncExecutionImpl.executeSync(SyncExecutionImpl.java:187)
kestra-1    |   at dev.failsafe.FailsafeExecutor.call(FailsafeExecutor.java:376)
kestra-1    |   at dev.failsafe.FailsafeExecutor.get(FailsafeExecutor.java:112)
kestra-1    |   at io.kestra.core.utils.RetryUtils$Instance.wrap(RetryUtils.java:144)
kestra-1    |   at io.kestra.core.utils.RetryUtils$Instance.runRetryIf(RetryUtils.java:103)
kestra-1    |   at io.kestra.jdbc.JooqDSLContextWrapper.transaction(JooqDSLContextWrapper.java:55)
kestra-1    |   at io.kestra.jdbc.runner.AbstractJdbcMultipleConditionStorage.save(AbstractJdbcMultipleConditionStorage.java:69)
kestra-1    |   at io.kestra.core.services.FlowTriggerService.computeExecutionsFromFlowTriggers(FlowTriggerService.java:114)
kestra-1    |   at io.kestra.jdbc.runner.JdbcExecutor.toExecution(JdbcExecutor.java:903)
kestra-1    |   at io.kestra.jdbc.runner.JdbcExecutor.toExecution(JdbcExecutor.java:864)
kestra-1    |   at io.kestra.jdbc.runner.JdbcExecutor.executionQueue(JdbcExecutor.java:570)
kestra-1    |   at java.base/java.util.ArrayList.forEach(Unknown Source)
kestra-1    |   at io.kestra.jdbc.runner.JdbcQueue.lambda$receive$7(JdbcQueue.java:264)
kestra-1    |   at io.kestra.jdbc.runner.JdbcQueue.lambda$receiveImpl$11(JdbcQueue.java:333)
kestra-1    |   at io.kestra.jdbc.runner.JdbcQueue.lambda$poll$12(JdbcQueue.java:356)
kestra-1    |   at io.micrometer.core.instrument.internal.TimedRunnable.run(TimedRunnable.java:49)
kestra-1    |   at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
kestra-1    |   at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
kestra-1    |   at java.base/java.lang.Thread.run(Unknown Source)
kestra-1    | Caused by: org.postgresql.util.PSQLException: ERROR: null value in column "condition_id" of relation "multipleconditions" violates not-null constraint
kestra-1    |   Detail: Failing row contains (company.team_silver_layer, {"end": "2024-12-02T15:00:00.000Z", "start": "2024-12-02T00:00:0..., company.team, silver_layer, null, 2024-12-02 00:00:00+00, 2024-12-02 15:00:00+00, null).
kestra-1    |   at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2733)
kestra-1    |   at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2420)
kestra-1    |   at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:372)
kestra-1    |   at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:517)
kestra-1    |   at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:434)
kestra-1    |   at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:194)
kestra-1    |   at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:180)
kestra-1    |   at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
kestra-1    |   at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
kestra-1    |   at org.jooq.tools.jdbc.DefaultPreparedStatement.execute(DefaultPreparedStatement.java:219)
kestra-1    |   at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:458)
kestra-1    |   at org.jooq.impl.AbstractDMLQuery.execute(AbstractDMLQuery.java:1068)
kestra-1    |   at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:348)
kestra-1    |   ... 38 common frames omitted
kestra-1    | 2024-12-02 14:51:11,613 INFO  default-nioEventLoopGroup-1-4 io.kestra.webserver.access 2024-12-02T14:51:09.720Z | GET /api/v1/executions/KxMpiT06Rxb2RH9CR3aIe/follow HTTP/1.1 | status: 200 | ip: 192.168.65.1 | length: 3123 | duration: 1892
kestra-1    | 2024-12-02 14:51:12,137 INFO  default-nioEventLoopGroup-1-4 io.kestra.webserver.access 2024-12-02T14:51:12.120Z | GET /api/v1/logs/KxMpiT06Rxb2RH9CR3aIe?minLevel=TRACE HTTP/1.1 | status: 200 | ip: 192.168.65.1 | length: 2 | duration: 17
kestra-1    | 2024-12-02 14:51:12,639 INFO  jdbc-queue-Execution_0 io.kestra.core.runners.Worker All working threads are terminated.
kestra-1    | 2024-12-02 14:51:12,674 WARN  jdbc-queue-Execution_0 i.k.c.server.ServiceLivenessManager Failed to transition service [id=6ldIYZI1ZQZAdHhIjArCQJ, type=INDEXER, hostname=37682afdda51] from TERMINATED_GRACEFULLY to TERMINATING. Cause: Invalid transition.
kestra-1    | 2024-12-02 14:51:12,674 ERROR jdbc-queue-Execution_0 i.k.c.server.ServiceLivenessManager [Service id=6ldIYZI1ZQZAdHhIjArCQJ, type=INDEXER, hostname=37682afdda51] Termination already completed (TERMINATED_GRACEFULLY). This error may occur if the service has already been evicted by Kestra due to a prior error.
kestra-1    | 2024-12-02 14:51:12,732 INFO  command-shutdown i.kestra.core.contexts.KestraContext Kestra server - Shutdown initiated
kestra-1    | 2024-12-02 14:51:12,732 INFO  command-shutdown i.kestra.core.contexts.KestraContext Kestra server - Shutdown completed
kestra-1    | 2024-12-02 14:51:12,732 WARN  command-shutdown io.kestra.cli.AbstractCommand Receiving shutdown ! Try to graceful exit
kestra-1    | 2024-12-02 14:51:12,732 WARN  command-shutdown io.kestra.cli.AbstractCommand Receiving shutdown ! Try to graceful exit
kestra-1 exited with code 1
```

### Environment

- Kestra Version: develop EE. 

Trying to recreate and add more detail

--- a/core/src/main/java/io/kestra/plugin/core/trigger/Flow.java
+++ b/core/src/main/java/io/kestra/plugin/core/trigger/Flow.java
@@ -124,7 +124,7 @@ import static io.kestra.core.utils.Rethrow.throwPredicate;
             full = true,
             title = """
                 2) Trigger the `silver_layer` flow once the `bronze_layer` flow finishes successfully by 9 AM.
-                
+
                 ```yaml
                 id: bronze_layer
                 namespace: company.team
@@ -132,7 +132,7 @@ import static io.kestra.core.utils.Rethrow.throwPredicate;
                 tasks:
                   - id: raw_data
                     type: io.kestra.plugin.core.log.Log
-                    message: Ingesting raw data                
+                    message: Ingesting raw data
                 ```""",
             code = """
                 id: silver_layer
@@ -221,6 +221,7 @@ public class Flow extends AbstractTrigger implements TriggerOutput<Flow.Output> 
     @Builder.Default
     private List<State.Type> states = State.Type.terminatedTypes();
 
+    @Valid
     @Schema(
         title = "Preconditions on upstream flow executions",
         description = "Express preconditions to be met, on a time window, for the flow trigger to be evaluated."
@@ -306,7 +307,6 @@ public class Flow extends AbstractTrigger implements TriggerOutput<Flow.Output> 
             You can disable this by setting this property to `false`, so that within the same window, each time one of the conditions is satisfied again after a successful evaluation, it will trigger a new execution."""
         )
         @PluginProperty
-        @NotNull
         @Builder.Default
         private Boolean resetOnSuccess = Boolean.TRUE;
 
@@ -415,7 +415,6 @@ public class Flow extends AbstractTrigger implements TriggerOutput<Flow.Output> 
         @Schema(title = "A unique identifier for the filter.")
         private String id;
 
-        @NotNull
         @Builder.Default
         @PluginProperty
         @Schema(title = "The operand to apply between all filters of the precondition.")

